components/fonts/font_context.rs

```diff
@@ -792,10 +820,20 @@ impl RemoteWebFontDownloader {
             None => return,
         };
 
-        // FIXME: This shouldn't use NoReferrer, but the current documents url
-        let request =
-            RequestBuilder::new(state.webview_id, url.clone().into(), Referrer::NoReferrer)
-                .destination(Destination::Font);
+        let document_context = &state.document_context;
+
+        let request = RequestBuilder::new(
+            state.webview_id,
+            url.clone().into(),
+            Referrer::ReferrerUrl(document_context.document_url.clone()),
+        )
+        .destination(Destination::Font)
+        .mode(RequestMode::CorsMode)
+        .credentials_mode(CredentialsMode::CredentialsSameOrigin)
+        .service_workers_mode(ServiceWorkersMode::All)
```

Comment:
If these are the default values, maybe they don't need to be set explicitly here?